src/Microsoft.DotNet.Darc/DarcLib/AzureDevOpsClient.cs

```diff
@@ -47,24 +49,25 @@ public class AzureDevOpsClient : RemoteRepoBase, IRemoteGitRepo, IAzureDevOpsCli
 
     // Azure DevOps uses this id when creating a new branch as well as when deleting a branch
     private static readonly string BaseObjectId = "0000000000000000000000000000000000000000";
+
+    private readonly IAzureDevOpsTokenProvider _tokenProvider;
     private readonly ILogger _logger;
-    private readonly string _personalAccessToken;
     private readonly JsonSerializerSettings _serializerSettings;
 
-    /// <summary>
-    /// Create a new azure devops client.
-    /// </summary>
-    /// <param name="accessToken">
-    ///     PAT for Azure DevOps. This PAT should cover all
-    ///     organizations that may be accessed in a single operation.
-    /// </param>
-    /// <remarks>
-    ///     The AzureDevopsClient currently does not utilize the memory cache
-    /// </remarks>
-    public AzureDevOpsClient(string gitExecutable, string accessToken, ILogger logger, string temporaryRepositoryPath)
-        : base(gitExecutable, temporaryRepositoryPath, null, logger, new RemoteTokenProvider(azureDevOpsToken: accessToken))
+    public AzureDevOpsClient(IAzureDevOpsTokenProvider tokenProvider, IProcessManager processManager, ILogger logger)
+        : this(tokenProvider, processManager, logger, null)
     {
-        _personalAccessToken = accessToken;
+    }
+
+    public AzureDevOpsClient(IAzureDevOpsTokenProvider tokenProvider, IProcessManager processManager, ILogger<AzureDevOpsClient> logger)
+        : this(tokenProvider, processManager, (ILogger)logger)
+    {
+    }
```

Comment:
It looks like this change caused a regression when DI instantiates the type with given constructor arguments. See https://github.com/dotnet/dnceng/issues/3380

Not sure why this pops up now and not earlier. Might be related to the SDK upgrade to P6.